testsuite/features/step_definitions/api_common.rb

```diff
@@ -0,0 +1,622 @@
+# Copyright (c) 2015-2022 SUSE LLC
+# Licensed under the terms of the MIT license.
+
+require 'json'
+require 'socket'
+
+$api_test = APITestHTTP.new($server.full_hostname)
+
+## auth namespace
+
+When(/^I am logged in API as user "([^"]*)" and password "([^"]*)"$/) do |user, password|
+  $api_test.auth_login(user, password)
+end
```

Comment:
I would go ahead and just login directly, not in a step, but just bellow the line 
``` 
$api_test = APITestHTTP.new($server.full_hostname)
```

Reply:
See answer before. The test object and the connecton can stay during the whole test suite, but I think you have to re-login every while and then. It's same problem than `Given I am connected as user ...` for UI.

If this is a wrong assumption we can do as you say.

Reply:
It would be good to have a longer keep-alive-timeout